rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java

```diff
@@ -95,21 +97,20 @@ public String getOrganizationName() {
         if (origin == null) {
             return null;
         }
-        String path;
-        if (origin.startsWith("git")) {
-            path = origin.substring(origin.indexOf(':') + 1);
-        } else {
-            path = URI.create(origin).getPath().substring(1);
-        }
-        int firstSlashPos = path.lastIndexOf('/');
-        int secondSlashPos = path.lastIndexOf('/', firstSlashPos - 1);
-
-        if (secondSlashPos > -1) {
-            return path.substring(secondSlashPos + 1, firstSlashPos);
-        } else if (firstSlashPos > -1) {
-            return path.substring(0, firstSlashPos);
-        } else {
-            return "";
+        try {
+            URIish gituri = new URIish(origin);
+            String path = gituri.getPath();
+            if (path.startsWith("/")) {
+                path = path.substring(1);
+            }
+            int firstIndexOfSlash = path.indexOf('/');
+            int lastIndexOfSlash = path.lastIndexOf('/');
+            if (firstIndexOfSlash != lastIndexOfSlash) {
```

Comment:
This isn't the best, as it fails on additional slashes such as:
`http://localhost:7990/scm/some/openrewrite/rewrite.git`
Those cases now also already requite a baseUrl, and are common in GitLab.